packages/next-app/src/app/internalmanager/(components)/InternalManagerProvider.tsx

```diff
@@ -0,0 +1,20 @@
+"use client";
+
+import { useRouter } from "next/navigation";
+import { useEffect } from "react";
+import { useAccount, useNetwork } from "wagmi";
+
+import WalletNotConnected from "#/components/WalletNotConnected";
+
+export function InternalManagerProvider({ children }: React.PropsWithChildren) {
+  const { isConnected } = useAccount();
+
+  const { chain } = useNetwork();
+  const { push } = useRouter();
+
+  useEffect(() => {
+    if (chain) push("/internalmanager");
+  }, [chain]);
+
+  return <>{isConnected ? <>{children}</> : <WalletNotConnected />}</>;
+}
```

Comment:
`#curious` is there a situation in which the wallet is connected but there's no chain? Alternatively, if there's a chain but the wallet is not connected it'll push you to `/internalmanager`. Is that the expected behavior here? I think you probably only want to depend on `chain` right?